demisto_sdk/commands/prepare_content/tests/yml_unifier_test.py

```diff
@@ -561,33 +561,77 @@ def mocked_get_api_module_code(*args, **kwargs):
-     - calling insert_pack_version.
+     - Calling insert_pack_version for different script types (.ps1, .js).
 
     Then:
-     - Ensure the code returned contains the pack version in it.
+     - Ensure the code returned contains the correct pack version.
+     - Ensure the pack version is added only once.
+     - Ensure unsupported script types do not get the pack version added.
     """
-    version_str = "### pack version: 1.0.3"
-    assert version_str not in DUMMY_SCRIPT
-    assert version_str in IntegrationScriptUnifier.insert_pack_version(
-        ".py", DUMMY_SCRIPT, "1.0.3"
-    )
-    assert version_str in IntegrationScriptUnifier.insert_pack_version(
-        ".ps1", DUMMY_SCRIPT, "1.0.3"
+    pack_version = "1.0.3"
+    version_str_ps1 = f"### pack version: {pack_version}"
+    version_str_js = f"// pack version: {pack_version}"
+
+    # Test for JavaScript script
+    updated_script_js = IntegrationScriptUnifier.insert_pack_version(
+        ".js", DUMMY_SCRIPT, pack_version
     )
-    assert version_str not in IntegrationScriptUnifier.insert_pack_version(
-        ".js", DUMMY_SCRIPT, "1.0.3"
+    assert version_str_js in updated_script_js
+    assert updated_script_js.count(version_str_js) == 1  # Ensure it is added only once
+
+    # Test for PowerShell script
+    updated_script_ps1 = IntegrationScriptUnifier.insert_pack_version(
+        ".ps1", DUMMY_SCRIPT, pack_version
     )
-    version_str_js = "// pack version: 1.0.3"
-    assert version_str_js not in DUMMY_SCRIPT
-    assert version_str_js in IntegrationScriptUnifier.insert_pack_version(
-        ".js", DUMMY_SCRIPT, "1.0.3"
+    assert version_str_ps1 in updated_script_ps1
+    assert (
+        updated_script_ps1.count(version_str_ps1) == 1
+    )  # Ensure it is added only once
+
+
+@pytest.mark.parametrize(
+    "dummy_script, pack_version, expected_version_str",
+    [
+        pytest.param(
+            """
+            def main():
+            """,
+            "1.0.3",
+            "demisto.debug('pack version = 1.0.3')",
+            id="script without version",
+        ),
+        pytest.param(
+            """
+            demisto.debug('pack version = 1.0.3')
+            def main():
+            """,
+            "1.0.4",
+            "demisto.debug('pack version = 1.0.4')",
+            id="script with version",
+        ),
+    ],
+)
+def test_insert_pack_version_and_script_to_yml_python_script(
+    dummy_script: str, pack_version: str, expected_version_str: str
+):
+    """
+    Test that the pack version is correctly inserted into the Python script.
+    """
+    updated_script = IntegrationScriptUnifier.insert_pack_version(
+        ".py", dummy_script, pack_version
     )
+    assert (
+        expected_version_str in updated_script
+    ), "Pack version string not found in the updated script"
+    assert (
+        updated_script.count(expected_version_str) == 1
+    ), "Pack version string found more than once in the updated script"
```

Comment:
can do something like `pack version found {...} times in updated script, expected only once`